### ISSUE TITLE
We need a mechanism for users to request experiments they want supported

### ISSUE THREAD

**Issue by jaclyn-taroni** [AlexsLemonade/refinebio#1056]
**TL;DR:** There may be an experiment that is highly relevant to our mission that has `no downloadable samples`, how can users let us know that they would like those data?

---

### Context

During the shepherding of 🐟 RNA-seq data, @kurtwheeler noticed that some of the jobs were failing because we did not have transcriptome indices for some of the organisms (e.g., _Arabidopsis thaliana_) included in a given experiment (e.g., `SRPXXXXX`). That means that some of the zebrafish experiments are in fact multispecies experiments. 

@kurtwheeler sent this my way so we could get an idea of how frequently multispecies RNA-seq experiments occur:

|   organism_names     | count |
|---------------------| ------- |
| {AMBYSTOMA_MEXICANUM,MUS_MUSCULUS,DANIO_RERIO}                                                                                                         |     1 |
 {DANIO_RERIO}                                                                                                                                          |   479
 {DROSOPHILA_MELANOGASTER,SACCHAROMYCES_CEREVISIAE,MUS_MUSCULUS,XENOPUS_LAEVIS,ARABIDOPSIS_THALIANA,HOMO_SAPIENS,SCHIZOSACCHAROMYCES_POMBE,DANIO_RERIO} |     1
 {GALLUS_GALLUS,HOMO_SAPIENS,DANIO_RERIO}                                                                                                               |     1
 {GALLUS_GALLUS,MUS_MUSCULUS,DANIO_RERIO}                                                                                                               |     1
 {GALLUS_GALLUS,MUS_MUSCULUS,XENOPUS_LAEVIS,PELODISCUS_SINENSIS_JAPONICUS,XENOPUS_TROPICALIS,CIONA_INTESTINALIS,BRANCHIOSTOMA_FLORIDAE,DANIO_RERIO}     |     1
 {HOMO_SAPIENS,DANIO_RERIO}                                                                                                                             |     3
 {MUS_MUSCULUS,DANIO_RERIO}                                                                                                                             |     6
 {OREOCHROMIS_NILOTICUS,DANIO_RERIO}                                                                                                                    |     1
 {RANA_CATESBEIANA,DANIO_RERIO}                                                                                                                         |     1
 {STRONGYLOCENTROTUS_PURPURATUS,DANIO_RERIO}                                                                                                            |     1

So for the zebrafish example where there are just shy of 500 experiments, ~3.5% of them of are multispecies. (Side note: my suspicion is that this may be _inflated_ compared to all of SRA because this is a model organism, but I have no data to back this up.)

We concluded that **for the moment we will not support multispecies RNA-seq experiments** because it would require us to subset experiments by organism to get to the `tximport` step and it happens infrequently enough that putting effort towards that right now seems imprudent.

Imagine, though, that one of the three `{HOMO_SAPIENS,DANIO_RERIO}` experiments was related to osteosarcoma or some other condition that's specifically part of our mission and an osteosarcoma researcher wants these data -- if it is displayed on the web UI because we have the associated metadata but there are no downloadable samples, how can the user quickly and easily bring this to our attention?

This is not specific to the multispecies RNA-seq experiment case! This could also happen with an Illumina BeadArray experiment that we're unable to process because the header of Illumina BeadArray raw data are not really standardized.

### Problem or idea

Users should be able to easily request experiments where there are no downloadable samples.

### Solution or next step

I don't yet have thoughts on how this would work or how it would be implemented, but I wanted to document this specific example to provide motivation for the enhancement.

@kurtwheeler brought up a good point: We need to be cognizant of _the process_ for identifying what experiments/edge cases we'd like to support. My initial thought was to use the _frequency_ of support requests to prioritize experiments, but this may prove insufficient.

### New Issue Checklist

- [x] The title is short and descriptive.
- [x] You have explained the context that led you to write this issue.
- [x] You have reported a problem or idea.
- [ ] You have proposed a solution or next step.

**Comment by dvenprasad** [comment]
But could we not process experiments with the Big 3 (Mouse, Human, and 🐟 )? It's 9 experiments + human,mouse experiments.

We can still track requests for other experiments. Based on the use case @jaclyn-taroni presented, we would also be interested in tracking why they interested in an experiment. Is that correct?

**Comment by kurtwheeler** [comment]
> But could we not process experiments with the Big 3

Yes we can. However it's a matter of it being worth the effort:

> We concluded that **for the moment we will not support multispecies RNA-seq experiments** because it would require us to subset experiments by organism to get to the `tximport` step and it happens infrequently enough that putting effort towards that right now seems imprudent.

**Comment by Miserlou** [comment]
Similarly, search results with no results found.

**Comment by cgreene** [comment]
@Miserlou makes a good point - we want folks to surface things like #1078 to us when something in the search isn't working.

**Comment by dvenprasad** [comment]
## **"Whiteboarding" (We tried a different method, Sketching, this time) and Low-Fi**

We identified 3 cases when the user might need to request for experiment(s)
### Consolidated sketches

**Case 1: Request an experiment which hasn't been processed**


![request-experiment](https://user-images.githubusercontent.com/15315514/54040410-4aa78000-4193-11e9-90bd-e913d2788f6a.jpg)

![request-experiment-form](https://user-images.githubusercontent.com/15315514/54040428-5430e800-4193-11e9-9683-f849fda687b2.png)

![redirect-experiment](https://user-images.githubusercontent.com/15315514/54040470-76c30100-4193-11e9-9670-1b3f937e183b.jpg)


**Case 2: Request for experiment(s) for search term when there are NO results**

![no-results](https://user-images.githubusercontent.com/15315514/54040548-ac67ea00-4193-11e9-9e35-7ed7ca851f01.png)

![request-missing-experiments-form](https://user-images.githubusercontent.com/15315514/54040610-cbff1280-4193-11e9-9454-9a0afbe9ee69.jpg)

![redirect-landing-page](https://user-images.githubusercontent.com/15315514/54040637-dd481f00-4193-11e9-97ec-9f83c0a6ff45.jpg)


**Case 3: Request experiment(s) for search term with results**
![search-term-with-results](https://user-images.githubusercontent.com/15315514/54040782-2d26e600-4194-11e9-82d1-be91b7c596a6.png)

They would be taken to same form in case 2. After submitting, the user will be redirected back to the results page. 

**Concerns:**

One of the major concerns was that in our attempt to understand who is making a request and what their research is, that the form would come off a little invasive. 


## High-Fi

@jaclyn-taroni and I worked on the questions little more. We decided to directly ask them whether or not they study pediatric cancer. We can monitor the responses and test our assumption that people may lie to get the experiment(s). If we do find that people are lying, we can modify the question accordingly. 

Alt solution (which requires a little more on dev side): Provide a type-ahead textbox, where they can type out the disease/condition they are studying. The options are populated based on an appropriate ontology. (@jaclyn-taroni correct me if this is not in line with what we discussed) 

**Case 1: Request an experiment which hasn't been processed**

![screen shot 2019-03-08 at 11 41 34 am](https://user-images.githubusercontent.com/15315514/54043176-f6ec6500-4199-11e9-80b0-e5c6c1a73a10.png)

![request-experiment-illustration](https://user-images.githubusercontent.com/15315514/54043201-05d31780-419a-11e9-8f65-e34813279fe8.png)

Alt: 
![request-experiment-sans-illustration](https://user-images.githubusercontent.com/15315514/54043217-0cfa2580-419a-11e9-8854-5fe363d08824.png)

![screen shot 2019-03-08 at 11 55 57 am](https://user-images.githubusercontent.com/15315514/54043262-24d1a980-419a-11e9-925b-76897dba7302.png)


**Case 2: Request for experiment(s) for search term when there are NO results**

![no-search-results](https://user-images.githubusercontent.com/15315514/54046295-ecce6480-41a1-11e9-8ba9-778295cc1324.png)

![request-experiment-illustration](https://user-images.githubusercontent.com/15315514/54047133-3b7cfe00-41a4-11e9-9a22-3c9252524908.png)


Alt:
![request-missing-results-sans-illustration](https://user-images.githubusercontent.com/15315514/54046326-ffe13480-41a1-11e9-899b-85bd7e712134.png)

![landing-page-confirmation](https://user-images.githubusercontent.com/15315514/54046337-02dc2500-41a2-11e9-8f89-6bfe5ebfc709.png)

**Case 3: Request experiment(s) for search term with results**

![screen shot 2019-03-08 at 1 04 46 pm](https://user-images.githubusercontent.com/15315514/54046565-cb21ad00-41a2-11e9-8420-465616a800aa.png)

User will be presented the same form above in case 2. 

(Redirects back to the search results after user submits the form)
![screen shot 2019-03-08 at 1 04 20 pm](https://user-images.githubusercontent.com/15315514/54046572-ceb53400-41a2-11e9-9766-da2569c98240.png)


Let me know if I need to make any tweaks.

**Comment by cgreene** [comment]
@dvenprasad : is this ready to go? When it is, we should close this & get a frontend ticket created for implementation.

**Comment by dvenprasad** [comment]
Invision Links:
https://invis.io/JPH5APXF68D

All related mockups are under Forms- #1056 section

Illustrations in assets/forms-illustration.
